compiler/src/test/scala/org/scalajs/core/compiler/test/JSInteropTest.scala

```diff
@@ -1541,6 +1688,467 @@ class JSInteropTest extends DirectTest with TestHelpers {
+      |def foo(x: Int): Int in trait B with JSName 'foo'
+      |
+      |      def foo(x: Int): Int
+      |          ^
+      |newSource1.scala:20: warning: A member of a JS class is overriding another member with a different JS name.
+      |
+      |override def foo(x: Int): Int in class C with JSName 'foo'
+      |    is conflicting with
+      |def foo(x: Int): Int in class A with JSName 'Syms.sym1'
+      |
+      |      override def foo(x: Int): Int = x
+      |                   ^
+    """
+
+    """
+    object Syms {
+      val sym1 = js.Symbol()
+    }
+
+    @ScalaJSDefined
+    class A[T] extends js.Object {
+      def foo(x: T): T = x
+    }
+    @ScalaJSDefined
+    trait B extends A[Int] {
+      @JSName(Syms.sym1)
+      def foo(x: Int): Int
+    }
+    @ScalaJSDefined
+    class C extends B {
+      override def foo(x: Int): Int = x
+    }
+    """ hasWarns
+    """
+      |newSource1.scala:16: warning: A member of a JS class is overriding another member with a different JS name.
+      |
+      |def foo(x: Int): Int in class A with JSName 'foo'
+      |    is conflicting with
+      |def foo(x: Int): Int in trait B with JSName 'Syms.sym1'
+      |
+      |      def foo(x: Int): Int
+      |          ^
+      |newSource1.scala:20: warning: A member of a JS class is overriding another member with a different JS name.
+      |
+      |override def foo(x: Int): Int in class C with JSName 'foo'
+      |    is conflicting with
+      |def foo(x: Int): Int in trait B with JSName 'Syms.sym1'
+      |
+      |      override def foo(x: Int): Int = x
+      |                   ^
+    """
+
+    """
+    object Syms {
+      val sym1 = js.Symbol()
+    }
+
+    @ScalaJSDefined
+    trait A extends js.Object {
+      def foo: Int
+    }
+    @ScalaJSDefined
+    trait B extends js.Object {
+      @JSName(Syms.sym1)
+      def foo: Int
+    }
+    @ScalaJSDefined
+    trait C extends A with B
+    """ hasWarns
+    """
+      |newSource1.scala:19: warning: A member of a JS class is overriding another member with a different JS name.
+      |
+      |def foo: Int in trait B with JSName 'Syms.sym1'
+      |    is conflicting with
+      |def foo: Int in trait A with JSName 'foo'
+      |
+      |    trait C extends A with B
+      |          ^
+    """
+
+    """
+    object Syms {
+      val sym1 = js.Symbol()
+    }
+
+    @ScalaJSDefined
+    trait A extends js.Object {
+      def foo: Int
+    }
+    @ScalaJSDefined
+    trait B extends js.Object {
+      @JSName(Syms.sym1)
+      def foo: Int
+    }
+    @ScalaJSDefined
+    abstract class C extends A with B
+    """ hasWarns
+    """
+      |newSource1.scala:19: warning: A member of a JS class is overriding another member with a different JS name.
+      |
+      |def foo: Int in trait B with JSName 'Syms.sym1'
+      |    is conflicting with
+      |def foo: Int in trait A with JSName 'foo'
+      |
+      |    abstract class C extends A with B
+      |                   ^
+    """
```

Comment:
I never realized that this is only a warning. Will this become an error? If no, we should:

- Have a valid use case for it.
- A way to silence the warning.

Both of these are of course completely orthogonal to this PR.